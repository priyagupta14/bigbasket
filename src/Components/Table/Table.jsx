/* eslint-disable react/prop-types */
import React from 'react';
// import PropTypes from 'prop-types';

const Table = (props) => {
  const { productList } = props;
  return (
    <table className="cart-table" border="1">
      <thead>
        <tr className="cart-table-header">
          <th>ITEM DESCRIPTION</th>
          <th>UNIT PRICE</th>
          <th>QUANTITY</th>
          <th>SUBTOTAL</th>
        </tr>
      </thead>
      {Object.keys(productList).map((category) => (
        <tbody key={category}>
          <tr className="title"><td>{category}</td></tr>
          {productList[category].map((product) => (
            <tr className="item-value" key={category + product.id}>
              <td>{product.name}</td>
              <td>{product.price}</td>
              <td>{(product.inCartCount) ? product.inCartCount : product.count }</td>
              <td>
                {((product.inCartCount)
                  ? product.inCartCount : product.count) * product.price}

              </td>
            </tr>
          ))}
        </tbody>
      ))}

    </table>
  );
};
export default Table;

// Table.propTypes = {
//   productList: PropTypes.shape({
//     category: PropTypes.arrayOf(
//       PropTypes.shape({
//         inCartCount: PropTypes.number.isRequired,
//         price: PropTypes.number.isRequired,
//         name: PropTypes.string.isRequired,
//       }),
//     ),
//   }).isRequired,
// };
