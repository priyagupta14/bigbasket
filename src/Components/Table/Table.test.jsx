/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import {
  fireEvent, render, screen,
} from '@testing-library/react';
import Table from './Table';

describe(Table.name, () => {
  const mockProps = {
    productList: {
      abc: [{
        category: 'abc',
        count: 2,
        id: 5,
        inCartCount: 0,
        name: 'milk',
        price: 10,
      },
      ],
    },
  };
  it('should match the snapshot', () => {
    const { container } = render(<Table {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
  it('should display pastOrder header', () => {
    render(<Table {...mockProps} />);
    screen.getByText('ITEM DESCRIPTION');
    screen.getByText('UNIT PRICE');
    screen.getByText('QUANTITY');
    screen.getByText('SUBTOTAL');
  });
  it('should match the table rows', () => {
    render(<Table {...mockProps} />);
    screen.getByText('milk');
    screen.getByText('10');
    screen.getByText('2');
    screen.getByText('20');
  });
});
