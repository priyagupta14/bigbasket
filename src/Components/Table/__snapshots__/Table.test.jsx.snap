// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should match the snapshot 1`] = `
<div>
  <table
    border="1"
    class="cart-table"
  >
    <thead>
      <tr
        class="cart-table-header"
      >
        <th>
          ITEM DESCRIPTION
        </th>
        <th>
          UNIT PRICE
        </th>
        <th>
          QUANTITY
        </th>
        <th>
          SUBTOTAL
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="title"
      >
        <td>
          abc
        </td>
      </tr>
      <tr
        class="item-value"
      >
        <td>
          milk
        </td>
        <td>
          10
        </td>
        <td>
          2
        </td>
        <td>
          20
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
